pkg/controller/oneagentapm/oneagentapm_controller.go

```diff
@@ -97,8 +97,16 @@ func (r *ReconcileOneAgentAPM) Reconcile(request reconcile.Request) (reconcile.R
 		return reconcile.Result{}, errors.New(".spec.apiUrl is missing")
 	}
 
+	if instance.Spec.UseImmutableImage {
+		r.dtcReconciler.UpdateAPIToken = true
+	}
+
 	dtc, upd, err := r.dtcReconciler.Reconcile(context.TODO(), instance)
```

Comment:
Technically `r.dtcReconciler` is at risk of race conditions if multiple OneAgentAPM `Reconcile` calls are running in parallel. I think by default running reconciliation in parallel is disabled so it's not a problem now, but I'd suggest the changes below (in part to avoid changes on global objects),

```suggestion
	dtcRec := *r.dtcReconciler
	if instance.Spec.UseImmutableImage {
		dtcRec.UpdateAPIToken = true
	}

	dtc, upd, err := dtcRec.Reconcile(context.TODO(), instance)
```